src/dodal/devices/i10/id_apple2.py

```diff
@@ -0,0 +1,464 @@
+        with self.add_children_as_readables():
+            self.id = id
+            self.pgm = pgm
+        with self.add_children_as_readables(HintedSignal):
+            self.energy_offset = soft_signal_rw(float, initial_value=0)
+
+    @AsyncStatus.wrap
+    async def set(self, value: float) -> None:
+        LOGGER.info(f"Moving f{self.name} energy to {value}.")
+        await asyncio.gather(
+            self.id.set(value=value + await self.energy_offset.get_value()),
+            self.pgm.energy.set(value),
+        )
+
+
+class I10Apple2Pol(StandardReadable, Movable):
+    """
+    Compound device to set polorisation of ID.
+
+    Parameters
+    ----------
+    id:
+        An I10Apple2 device.
+    prefix:
+        Not in use but needed for device_instantiation.
+    name:
+        New device name.
+    """
+
+    def __init__(self, id: I10Apple2, prefix: str = "", name: str = "") -> None:
+        super().__init__(name=name)
+        with self.add_children_as_readables():
+            self.id = id
+
+    @AsyncStatus.wrap
+    async def set(self, value: str) -> None:
+        self.id.pol = value  # change polarisation.
+        LOGGER.info(f"Changing f{self.name} polarisation to {value}.")
+        await self.id.set(
+            await self.id.energy.get_value()
+        )  # Move id to new polarisation
+
+
+def convert_csv_to_lookup(
+    file: str,
+    source: tuple[str, str],
+    mode: str = "Mode",
+    min_energy: str = "MinEnergy",
+    max_energy: str = "MaxEnergy",
+    poly_deg: list | None = None,
+):
+    """
+    Convert csv to a dictionary that can be read by Apple2 ID device.
+
+    Parameters
+    -----------
+    file: str
+        File path.
+    source: tuple[str, str]
+        Tuple(column name, source name)
+        e.g. ("Source", "idu").
+    mode: str = "Mode"
+        Column name for the available modes, "lv","lh","pc","nc" etc
+    min_energy: str = "MinEnergy":
+        Column name for min energy for the polynomial.
+    max_energy: str = "MaxEnergy",
+        Column name for max energy for the polynomial.
+    poly_deg: list | None = None,
+        Column names for the parameters for the polynomial, starting with the least significant.
+
+    """
+    if poly_deg is None:
+        poly_deg = [
+            "7th-order",
+            "6th-order",
+            "5th-order",
+            "4th-order",
+            "3rd-order",
+            "2nd-order",
+            "1st-order",
+            "b",
+        ]
+    look_up_table = {}
+    pol = []
+
+    def data2dict(row):
+        # logical for the conversion for each row of data.
+        if row[mode] not in pol:
+            pol.append(row[mode])
+            look_up_table[row[mode]] = {}
+            look_up_table[row[mode]]["Energies"] = {}
+            look_up_table[row[mode]]["Limit"] = {}
+            look_up_table[row[mode]]["Limit"]["Minimum"] = float(row[min_energy])
+            look_up_table[row[mode]]["Limit"]["Maximum"] = float(row[max_energy])
```

Comment:
could we just set a new dictionary assignment here to look_up_table[row[mode]] = new_dict?